File: solutions/day01/solution.clj
(def moves (->> (slurp "input.txt")
                (map #(if (= \( %) 1 -1)))) 

(defn cumsum
  [s]
  (reduce
    (fn [cs x] 
      (conj cs (+ (or (last cs) 0) x)))
    []
    s))

(println "part 1:" (reduce + moves))
(println "part 2:" (inc (.indexOf (cumsum moves) -1)))

File: solutions/day01/solution.scala
val input = scala.io.Source.fromFile("input.txt").toList.map(x => if (x == '(') 1 else -1)
println(s"part 1: ${input.sum}")

val cumsum = input.map{var s = 0; d => {s += d; s}}
println(s"part 2: ${cumsum.indexOf(-1) + 1}")
